src/Types/UsedTxOuts.purs

```diff
@@ -109,18 +187,45 @@ unlockTxOutRefs txOutRefs' =
   in
     ask >>= (unwrap >>> Ref.modify_ updateCache >>> liftEffect)
 
+-- | Remove used marks from all inputs that are saved in a `TxOutRefUnlockKeys`.
+-- Use this on the result of a previous lockTransactionInputs
+unlockTxOutKeys
+  :: forall (m :: Type -> Type) (a :: Type)
+   . MonadAsk UsedTxOuts m
+  => MonadEffect m
+  => TxOutRefUnlockKeys
+  -> m Unit
+unlockTxOutKeys = unlockTxOutRefs <<< cacheToRefs <<< unwrap
+  where
+  cacheToRefs
+    :: TxOutRefCache
+    -> Array { transactionId :: TransactionHash, index :: UInt }
+  cacheToRefs cache = concatMap flatten $ Map.toUnfoldable cache
+
+  flatten
+    :: Tuple TransactionHash (Set UInt)
```

Comment:
```suggestion
    :: TransactionHash /\ Set UInt
```

To use that, you also need to replace line 41 with
```
import Data.Tuple.Nested (type (/\), (/\))
```
(see the comment below too)